atuin.nix

```
@@ -35,6 +35,12 @@ rustPlatform.buildRustPackage {
       --zsh <($out/bin/atuin gen-completions -s zsh)
   '';
 
+  # Additional flags passed to the cargo test binary, see `cargo test -- --help`
+  checkFlags = [
+    # Registration tests require a postgres server
+    "--skip=registration"
```

```suggestion
    "--lib --bins"
```

Or similar may work better? If that's exactly what is passed to `cargo test`, then it will ensure we do not run any integration tests here

---

Merging as-is for now, I'll look at changing or updating this when we add more integration tests!

---

The call to `cargo test` is [this invocation](https://github.com/NixOS/nixpkgs/blob/28482dfd4284560ccf20fb2d9535be326cac8d54/pkgs/build-support/rust/hooks/cargo-check-hook.sh#L36). The arguments passed to `checkFlags` are passed into the binary with `cargo test -- --lib --bins`, so that won't work, and it doesn't look like there's a way to pass those arguments without co-opting some other feature.

If that way of calling cargo makes more sense for the tests here, it's probably clearer to just override the `checkPhase` and write the testing script ourselves. It's just a bash script.

```
# Override the checkPhase script. This is run after building to verify correctness.
# Uses release mode to avoid building in debug mode just for testing
checkPhase = ''
  cargo test --release --lib --bins
'';
```

---

Ah I see, thanks for explaining! Will figure out which makes the most sense when we add more tests